src/main/java/de/tum/cit/aet/artemis/programming/repository/ProgrammingExerciseRepository.java

```diff
@@ -149,6 +149,25 @@ default ProgrammingExercise findOneByProjectKeyOrThrow(String projectKey, boolea
         return exercises.getFirst();
     }
 
+    /**
+     * Get a programmingExercise for a student, with the latest result and feedbacks.
+     *
+     * @param projectKey of the exercise that should be fetched.
+     * @return the exercise with the given ID, if found.
+     */
+    @Query("""
+            SELECT DISTINCT pe
+            FROM ProgrammingExercise pe
+                LEFT JOIN FETCH pe.studentParticipations sp
+                LEFT JOIN FETCH sp.results spr
```

Comment:
Navigating from the student participation to the result is deprecated and should not be used any more